# Obtaining and throwing an error with a custom message in the context of a field's key

-- input
field: value
-- input

-- javascript
throw enolib.parse(input).field('field').keyError('my message')
-- javascript

-- php
throw Enolib\Parser::parse($input)->field('field')->keyError('my message');
-- php

-- python
raise enolib.parse(input).field('field').key_error('my message')
-- python

-- ruby
raise Enolib.parse(input).field('field').key_error('my message')
-- ruby

## ValidationError

-- text
There is a problem with the key of this element: my message
-- text

-- snippet
   Line | Content
 >    1 | field: value
-- snippet

selection: [0,0] => [0,5]

# Obtaining and throwing an error with a custom generated message in the context of a field's key

-- input
field: value
-- input

-- javascript
throw enolib.parse(input).field('field').keyError(key => `my generated message for field '${key}'`)
-- javascript

-- php
throw Enolib\Parser::parse($input)->field('field')->keyError(function($key) { return "my generated message for field '{$key}'"; });
-- php

-- python
raise enolib.parse(input).field('field').key_error(lambda key: f"my generated message for field '{key}'")
-- python

-- ruby
raise Enolib.parse(input).field('field').key_error { |key| "my generated message for field '#{key}'" }
-- ruby

## ValidationError

-- text
There is a problem with the key of this element: my generated message for field 'field'
-- text

-- snippet
   Line | Content
 >    1 | field: value
-- snippet

selection: [0,0] => [0,5]
